packages/openchs-android/src/views/filter/FiltersView.js

```diff
@@ -144,6 +153,31 @@ class FilterView extends AbstractComponent {
         this.dispatchAction(FilterActionNames.ADD_PROGRAM, {programUUID: uuid});
     }
 
+    clearData() {
+        this.dispatchAction(FilterActionNames.ON_DATE_CLEAR,
+            {filterDate: this.state.filterDate})
+        for(len in this.state.selectedGeneralEncounterTypes){
+            this.dispatchAction(FilterActionNames.ADD_GENERAL_VISITS,
+                {encounterUUID: this.state.selectedGeneralEncounterTypes[len]['uuid']})}
+        var MaxAttempts = 3;
+        while (MaxAttempts>=0){
+            this.dispatchAction(FilterActionNames.INDIVIDUAL_SEARCH_ADDRESS_LEVEL_CLEAR, {
+                    addressLevelState: this.state.addressLevelState});
+            MaxAttempts--;}
+        for (len in this.state.selectedCustomFilters){
+            this.dispatchAction(CustomFilterNames.ON_CODED_CUSTOM_FILTER_CLEAR, [len])}
+        for(len in this.state.selectedEncounterTypes){
+            this.dispatchAction(FilterActionNames.ADD_VISITS, {encounterUUID: this.state.selectedEncounterTypes[len]['uuid']})}
+        for(len in this.state.selectedGenders){
+            gender = this.state.selectedGenders[len].name;
+            this.dispatchAction(Actions.ON_GENDER_SELECT,{gender})}
+        if (this.state.programs.length != 1)   {
+            for(len in this.state.selectedPrograms){
+                this.onProgramSelect(null,this.state.selectedPrograms[len].uuid)
+            }
```

Comment:
instead of dispatching multiple actions we should dispatch one action and move all this logic in FilterAction file.

Reply:
You mean, Like addProgram function we can add removeProgram function in FilterAction file
?

Reply:
When reset button is pressed we should dispatch single action (say RESET_DASHBOARD_FILTERS). And all this state resitting logic will go inside that action (in FilterAction.js you can have separate functions for each filter type).

Reply:
Okay...thanks for inputs.

Reply:
Done with changes.
Optimize the code.